src/libraries/System.Private.CoreLib/src/System/IO/UnmanagedMemoryAccessor.cs

```diff
@@ -494,8 +494,8 @@ public void Write(long position, decimal value)
                 int* valuePtr = (int*)(&value);
                 int flags = *valuePtr;
                 int hi = *(valuePtr + 1);
-                int lo = *(valuePtr + 2);
-                int mid = *(valuePtr + 3);
+                int lo = *(valuePtr + (BitConverter.IsLittleEndian ? 2 : 3));
+                int mid = *(valuePtr + (BitConverter.IsLittleEndian ? 3 : 2));
```

Comment:
Why is this the correct fix?  More generally, why is a fix needed here at all?  All the other methods are writing values using the machine endianness... why is decimal special?

Reply:
Because the internal implementation of decimal uses a long here, which is split into two halves by this cast to int.  Therefore the two halves end up swapped on big-endian.

Reply:
Does that mean you believe Decimal.{Try}GetBits is also broken?

Reply:
Ah, TryGetBits is actually accessing it as a long and shifting.

Can we just use TryGetBits here instead of propagating direct access to the decimal's state?

Reply:
Sure.   My initial patch used GetBits, which you didn't like due to the extra allocation.  I can try using TryGetBits instead, if that's OK performance-wise.  (However, that seems to require Span ... it that available in this file?  I had some issues in the past with CI failures due to introducing uses of Span.)

Reply:
Yup, span should be fine here, and you can write directly into the destination, e.g. something like:
```C#
_buffer.AcquirePointer(ref pointer);
pointer += (_offset + position);
decimal.TryGetBits(decimal, new Span<int>((int*)pointer, 4), out int intsWritten);
Debug.Assert(intsWritten == 4);
```

Reply:
Ah, I didn't see the edit before, sorry.   Just committed a patch to implement the original suggestion, which works for me.
I'll go ahead and update to use stackalloc.

Reply:
OK, now updated to use stackalloc.